cmd/erasure-healing.go

```diff
@@ -535,18 +536,18 @@ func (er erasureObjects) healObject(ctx context.Context, bucket string, object s
 		// record the index of the updated disks
 		partsMetadata[i].Erasure.Index = i + 1
```

Comment:
This shouldn't be done for Transitioned objects as this can change the erasure index order @krisis - this index should be only changed if its set to `0` i.e when fi.IsRemote() is false - this allows for Transitioned objects to be not touched.

Reply:
Unless the order is always maintained after the `shuffleDisks` then perhaps this is okay but it is still worth doing this only when Index is `0`

Reply:
> Unless the order is always maintained after the `shuffleDisks` then perhaps this is okay but it is still worth doing this only when Index is `0`

meta.Erasure.Index value in outdatedDisks are set to that of the 'latestMeta' (which is the first valid FileInfo value). So, even though the index is non-zero it is invalid. This value is important for both regular and transitioned objects alike.